File: SWEA/D4/1210. ［S／W 문제해결 기본］ 2일차 － Ladder1/［S／W 문제해결 기본］ 2일차 － Ladder1.py
for _ in range(10):
  tc = int(input())
  ladder = [ list(map(int, input().split())) for _ in range(100)]
  for i in range(100):
    if ladder[99][i] == 2:
      start = i
  y = 99    
  while True:
    if y == 0:
      break
    if 0 <= start-1 and ladder[y][start-1] == 1:
      while 0 <= start-1 and ladder[y][start-1] == 1:
        start -= 1
    elif start+1 <= 99 and ladder[y][start+1] == 1:
      while start+1 <= 99 and ladder[y][start+1] == 1:
        start += 1
    y -= 1
  print(f"#{tc} {start}")

File: SWEA/D4/1210. ［S／W 문제해결 기본］ 2일차 － Ladder1/［S／W 문제해결 기본］ 2일차 － Ladder1.java
import java.io.*;
import java.util.*;

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        for (int test = 0; test < 10; test++) {
            int tc = Integer.parseInt(br.readLine());
            int[][] ladder = new int[100][100];

            for (int i = 0; i < 100; i++) {
                StringTokenizer st = new StringTokenizer(br.readLine());
                for (int j = 0; j < 100; j++) {
                    ladder[i][j] = Integer.parseInt(st.nextToken());
                }
            }

            int start = 0;
            for (int i = 0; i < 100; i++) {
                if (ladder[99][i] == 2) {
                    start = i;
                    break;
                }
            }

            int y = 99;
            while (true) {
                if (y == 0) break;

                if (start - 1 >= 0 && ladder[y][start - 1] == 1) {
                    while (start - 1 >= 0 && ladder[y][start - 1] == 1) {
                        start--;
                    }
                } else if (start + 1 < 100 && ladder[y][start + 1] == 1) {
                    while (start + 1 < 100 && ladder[y][start + 1] == 1) {
                        start++;
                    }
                }

                y--;
            }

            System.out.println("#" + tc + " " + start);
        }
    }
}